lightning-invoice/src/lib.rs

```diff
@@ -970,6 +1036,7 @@ impl Invoice {
 			signed_invoice: signed_invoice,
 		};
 		invoice.check_field_counts()?;
+		invoice.check_feature_bits()?;
```

Comment:
Are there cases where you may want to use this on an invoice *not* generated by us? Where we may want to accept a non-basic-mpp invoice but not generate them?

Reply:
I believe this is supported as written. `check_feature_bits` simply checks that BOLT 11 requirements are met. Note that the changes in this PR don't require that any feature bits are set. I know we discussed doing that offline yesterday, but I ended up finding a simpler way to enforce the semantics.